pkg/vault/vault.go

```diff
@@ -106,14 +114,81 @@ func (v *Vault) ReadSecret(path string) ([]byte, error) {
 
 	data, ok := secret.Data["value"].(string)
 	if !ok {
-		return nil, fmt.Errorf("secret data type is not string, found %T value: %#v", secret.Data["value"], secret.Data["value"])
+		return nil, fmt.Errorf("secret data type is not string, found %T value: %#v at path: %s", secret.Data["value"], secret.Data["value"], path)
 	}
 
 	return []byte(data), nil
 }
 
+func (v *Vault) manageTokenLifecycle(ctx context.Context, authToken *hashivault.Secret) error {
+	authTokenWatcher, err := v.client.NewLifetimeWatcher(&hashivault.LifetimeWatcherInput{
+		Secret: v.token,
+	})
+	if err != nil {
+		return fmt.Errorf("error initializing auth token lifetime watcher: %v", err)
```

Comment:
This is the only error returned by this function. If we move initialization of watcher outside of this function, then `manageTokenLifecycle` doesn't need return value.